File: packages/typescript/src/endpoints/scan_rules.ts
/**
 * Scan_Rules API Client
 */

import { AxiosInstance } from 'axios'

export class Scan_RulesClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * POST /rules
   */
  async postRules(rule_in: ScanRuleCreate): Promise<ScanRuleSchema> {
        const params = {}
        const data = rule_in
        return this.httpClient.post(`/rules`, data, params)
  }
  /**
   * GET /rules
   */
  async getRules(skip?: number, limit?: number, tool_filter?: string, category_filter?: string, severity_filter?: string, language_filter?: string, is_active?: boolean, organization_id?: string): Promise<List[ScanRuleSchema]> {
        const params = {'skip: skip, limit: limit, tool_filter: tool_filter, category_filter: category_filter, severity_filter: severity_filter, language_filter: language_filter, is_active: is_active, organization_id: organization_id'}
        const data = undefined
        return this.httpClient.get(`/rules`, params)
  }
  /**
   * GET /rules/{rule_id}
   */
  async getRules{rule_id}(rule_id: string): Promise<ScanRuleSchema> {
        const params = {'rule_id: rule_id'}
        const data = undefined
        return this.httpClient.get(`/rules/{rule_id}`, params)
  }
  /**
   * POST /rules/upload
   */
  async postRulesupload(file?: UploadFile, organization_id?: string): Promise<any> {
        const params = {}
        const data = {'file: file, organization_id: organization_id'}
        return this.httpClient.post(`/rules/upload`, data, params)
  }
  /**
   * PUT /rules/{rule_id}
   */
  async putRules{rule_id}(rule_id: string, rule_update: ScanRuleUpdate): Promise<ScanRuleSchema> {
        const params = {}
        const data = {'rule_id: rule_id, rule_update: rule_update'}
        return this.httpClient.put(`/rules/{rule_id}`, data, params)
  }
  /**
   * DELETE /rules/{rule_id}
   */
  async deleteRules{rule_id}(rule_id: string): Promise<any> {
        const params = {}
        const data = rule_id
        return this.httpClient.delete(`/rules/{rule_id}`, data, params)
  }
}

File: packages/python/src/endpoints/scan_rules.py
"""
Scan_Rules API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class Scan_RulesClient:
    """Client for scan_rules endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def post_rules(self, rule_in: Any):
        """POST /rules"""
        url = f"/rules"
        params = {}
        data = rule_in
        return await self._request("POST", url, params=params, json=data)


    async def get_rules(self, skip: Optional[float] = None, limit: Optional[float] = None, tool_filter: Optional[str] = None, category_filter: Optional[str] = None, severity_filter: Optional[str] = None, language_filter: Optional[str] = None, is_active: Optional[bool] = None, organization_id: Optional[str] = None):
        """GET /rules"""
        url = f"/rules"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_rules_rule_id(self, rule_id: str):
        """GET /rules/{rule_id}"""
        url = f"/rules/{rule_id}"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_rules_upload(self, file: Optional[Any] = None, organization_id: Optional[str] = None):
        """POST /rules/upload"""
        url = f"/rules/upload"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def put_rules_rule_id(self, rule_id: str, rule_update: Any):
        """PUT /rules/{rule_id}"""
        url = f"/rules/{rule_id}"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("PUT", url, params=params, json=data)


    async def delete_rules_rule_id(self, rule_id: str):
        """DELETE /rules/{rule_id}"""
        url = f"/rules/{rule_id}"
        params = {}
        data = rule_id
        return await self._request("DELETE", url, params=params, json=data)

